### PR TITLE
Add Red Hat headers to all sources in Kogito

### PR DESCRIPTION
**Pull request by bxf12315**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -25,6 +25,9 @@
     <!-- containers used for testing -->
     <container.image.infinispan>quay.io/infinispan/server:${version.org.infinispan.image}</container.image.infinispan>
     <container.image.keycloak>quay.io/keycloak/keycloak:${version.org.keycloak.image}</container.image.keycloak>
+    <!-- license plugin -->
+    <version.license.plugin>3.0</version.license.plugin>
+    <skipLicensesCheck>false</skipLicensesCheck>
```
Comment (cristianonicolai) [inline pom.xml]: isnt skip a property from the plugin already? not sure we need to redefine it here.

File: pom.xml [hunk 2]
```diff
@@ -35,6 +38,42 @@
           <artifactId>frontend-maven-plugin</artifactId>
           <version>${version.frontend.plugin}</version>
         </plugin>
+        <plugin>
+          <groupId>com.mycila</groupId>
+          <artifactId>license-maven-plugin</artifactId>
+          <version>${version.license.plugin}</version>
```
Comment (cristianonicolai) [inline pom.xml]: please define the plugin version and common configuration in the parent pom: https://github.com/kiegroup/kogito-runtimes/blob/master/pom.xml#L935

File: licensesheader.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Copyright ${license.git.copyrightLastYear} Red Hat, Inc. and/or its affiliates.
```
Comment (cristianonicolai) [inline licensesheader.txt]: I think what we want is `license.git.copyrightCreationYear`

File: pom.xml (hunk 1 above)
Comment (bxf12315) [reply pom.xml]: I think it not already exits.

File: pom.xml (hunk 2 above)
Comment (bxf12315) [reply pom.xml]: For kogito-apps's plugin is different to kogito-runtimes. you can see kogito-runtimes https://github.com/Cory-jbpm/kogito-runtimes/blob/license-header/pom.xml#L1351.

File: licensesheader.txt (hunk 3 above)
Comment (bxf12315) [reply licensesheader.txt]: the param is not work in version 3.0, you can see here https://github.com/mycila/license-maven-plugin/blob/license-maven-plugin-3.0/license-maven-plugin-git/src/main/java/com/mycila/maven/plugin/license/git/CopyrightRangeProvider.java#L41. license.git.copyrightCreationYear is work in version 3.1-SNAPSHOT. I think we can not use a snapshot version.

Comment (cristianonicolai) [reply licensesheader.txt]: but last year from what I understood will force every file to be changed to last commit year. We usually only add the year when the file is created, then no need to update it again. Should we probably wait for it to be released and adjust kogito-examples?

File: pom.xml (hunk 2 above)
Comment (cristianonicolai) [reply pom.xml]: define the _common_ configuration there ;)

File: pom.xml (hunk 1 above)
Comment (cristianonicolai) [reply pom.xml]: `-Dlicense.skip=true` does not work?

File: licensesheader.txt (hunk 3 above)
Comment (bxf12315) [reply licensesheader.txt]: the plugin does not update for four years. I think use update time is reasonable, on the other hand, the plugin's format is different to our old one.

File: pom.xml (hunk 1 above)
Comment (bxf12315) [reply pom.xml]: yes it not work.

File: pom.xml (hunk 2 above)
Comment (bxf12315) [reply pom.xml]: ok

**Comment by bxf12315** [comment]
dependent on https://github.com/kiegroup/kogito-runtimes/pull/527

File: licensesheader.txt (hunk 3 above)
Comment (cristianonicolai) [reply licensesheader.txt]: using _copyrightLastYear_ will force to update the copywrite for all files we change next year or am I missing something?

Comment (bxf12315) [reply licensesheader.txt]: license.git.copyrightLastYear - the year of the last change of the present file as seen in git history.
So I think it's reasonable.

Comment (cristianonicolai) [reply licensesheader.txt]: I dont think this is the way we handle it, we add copywrite once the file is created, thats it, we dont keep changing it. I let other weight in, but doesnt seem the correct approach to me. @tiagodolphine @evacchi @ricardozanini wdyt?

**Comment by ricardozanini** [comment]
Needs rebase

**Comment by bxf12315** [comment]
@cristianonicolai Please hele me reopen it. Thanks

**Comment by bxf12315** [comment]
> Needs rebase
new PR:https://github.com/kiegroup/kogito-apps/pull/244

**Pull request closed without merging**